src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -122,9 +122,13 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
 
         // if password is set, it takes precedence over hash
         String plainTextPassword = additionalSettingsBuilder.get("password");
-        if (plainTextPassword != null && plainTextPassword.length() > 0) {
-            additionalSettingsBuilder.remove("password");
-            additionalSettingsBuilder.put("hash", hash(plainTextPassword.toCharArray()));
+        if (plainTextPassword != null) {
+            if (plainTextPassword.length() == 0)
+                additionalSettingsBuilder.remove("password");
```

Comment:
This statement is present in both "if" and "else".
We can remove from "if else" and just put before "if else"